### PR TITLE
clinic v0.8.0: Support collection of SQL query plan related information

### PR DESCRIPTION
**Pull request by Oreoxmt**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Support collection of SQL query plan related information in Clinic v0.8.0

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: clinic/clinic-collect-sql-query-plan.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+Clinic Diag 诊断客户端（下称 Diag）集成了 `PLAN REPLAYER` 功能，支持 TiDB v4.0 以上的集群。你可以使用 Diag 方便快捷地保存查询计划相关的数据。
+
+> **警告：**
+>
+> - 当前该功能为实验特性，不建议在生产环境中使用。
+> - 暂时不支持采集 TiDB Operator 部署集群的数据。
+
+- 导出排查现场 TiDB 集群的相关信息，导出为 ZIP 格式的文件用于保存。
+- Diag 中采集的数据与 TiDB 中 `PLAN REPLAYER` 功能采集的数据有少量差距，具体见[采集输出结果](#采集输出结果)。
```
Comment (Oreoxmt) [inline clinic/clinic-collect-sql-query-plan.md:20]: @qqqdan 采集输出信息部分并没有说明比 PLAN REPLAYER 少的采集数据是哪些

File: clinic/clinic-collect-sql-query-plan.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+TiDB 在 v5.3.0 中引入了 [`PLAN REPLAYER`](/sql-plan-replayer.md) 功能，能够一键式导入或导出重现查询计划相关的信息。`PLAN REPLAYER` 提供了足够简单、一站式获取并保留线上事故场景的功能，简化了优化器相关问题的排查信息提取步骤。
+
+## 使用说明
+
+Clinic Diag 诊断客户端（下称 Diag）集成了 `PLAN REPLAYER` 功能，支持 TiDB v4.0 以上的集群。你可以使用 Diag 方便快捷地保存查询计划相关的数据。
+
+> **警告：**
+>
+> - 当前该功能为实验特性，不建议在生产环境中使用。
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:16]: 为什么说是实验特性呀？

**Review by Yisaer — COMMENTED** [review]
rest lgtm

File: clinic/clinic-collect-sql-query-plan.md (hunk 2 above)
Comment (Oreoxmt) [reply clinic/clinic-collect-sql-query-plan.md:16]: > 为什么说是实验特性呀？

@qqqdan PTAL

**Review by qqqdan — APPROVED** [review]

**Review by qqqdan — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

File: clinic/clinic-collect-sql-query-plan.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+`sql-statement` 的文件内容说明：
+
+- 如果有多条 SQL 语句需要分析，可以使用分号（;）分隔。
+- 因为 `diag` 都是新建会话进行查询，所以 SQL 文件里的 SQL 语句需要指明所使用的数据库。
+
+#### 采集输出结果
+
+采集输出结果包含以下集群现场信息：
```
Comment (TomShawn) [inline clinic/clinic-collect-sql-query-plan.md:71]: 「采集的输出结果」还是「采集输出结果」呢？偏正结构还是动宾结构？意思不一样

Comment (TomShawn) [reply clinic/clinic-collect-sql-query-plan.md:71]: 如果下表是采集的数据，能否在文档一开始就说一下，Diag 能采集到哪些现场数据？可能也要在前面说一下，在使用 Diag 前用户需要准备好 SQL 文件（前置信息）。

File: clinic/clinic-collect-sql-query-plan.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    tiup diag --version
+    ```
+
+    如果 Diag 版本不满足要求，可以通过如下命令升级到最新版本：
+
+    ```bash
+    tiup update diag
+    ```
+
+### 采集数据
```
Comment (TomShawn) [inline clinic/clinic-collect-sql-query-plan.md:43]: 这其实是全量信息采集，与下面的自定义信息采集对应吧？这里能否强调下，是标准信息采集，或者全量信息采集呢？

File: clinic/clinic-collect-sql-query-plan.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+`sql-statement` 的文件内容说明：
+
+- 如果有多条 SQL 语句需要分析，可以使用分号（;）分隔。
+- 因为 `diag` 都是新建会话进行查询，所以 SQL 文件里的 SQL 语句需要指明所使用的数据库。
+
+#### 采集输出结果
+
+采集输出结果包含以下集群现场信息：
+
+| 序号 | 采集内容 | 调用的 Diag collector | 输出文件 |
```
[the remark this answers is not in the document]
Comment (Oreoxmt) [reply clinic/clinic-collect-sql-query-plan.md:75]: 只是方便表格好看，也可以不加编号

File: clinic/clinic-collect-sql-query-plan.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+```bash
+diag collect <cluster-name> --profile=tidb-plan-replayer --explain-sql=<statement-filepath>
+```
+
+> **注意：**
+>
+> - 通过 `diag` 采集数据前，你需要提供 `sql-statement` 文件，该文件包含了需要采集的 SQL 语句。
+> - 以上命令中 `<statement-filepath>` 是指 `sql-statement` 文件的路径。
+> - `PLAN REPLAYER` **不会**导出表中数据。
+> - 采集数据时只会执行 `EXPLAIN`，不会真正执行查询。因此采集时对数据库性能影响较小。
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:58]: 这里最好在增加 2 个注意点:
1. --explain-sql 支持多条 sql 查询，不同 sql 间用 ; 相隔
2. --explain-sql 中的表名必须显式指定 database, 如 `select * from test.t`

File: clinic/clinic-collect-sql-query-plan.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+`sql-statement` 的文件内容示例：
+
+```sql
+SELECT * FROM information_schema.slow_query;SELECT * FROM information_schema.statements_summary
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:63]: 这里最好不要用 infoschema 的表来做例子。

File: clinic/clinic-collect-sql-query-plan.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+| 序号 | 采集内容 | 调用的 Diag collector | 输出文件 |
+| :--- | :--- | :--- | :--- |
+| 1 | TiDB 配置信息 | `config` | `tidb.toml` |
+| 2 | TiDB Session 系统变量 | `db_vars` | `global_variables.csv`，`mysql.tidb.csv` |
+| 3 | TiDB 执行计划绑定信息 (SQL Binding) | `sql_bind` | `sql_bind/global_bind.csv` |
+| 4 | `sql-statement` 中所包含的表结构 | `statistics` | `statistics/<db.table>.schema` |
+| 5 | `sql-statement` 中所包含表的统计信息 | `statistics` | `statistics/<db.table>.json` |
+| 6 | `EXPLAIN [ANALYZE] sql-statement` 的结果 | `explain` | `explain/sql0` |
+| 7 | <ul><li>TiDB 日志</li><li>TiUP Cluster 操作日志</li></ul> | <ul><li>`log`</li><li>`-R=tidb`</li></ul> | `tidb.log`，`tidb_slow_query.log`，`tidb_stdeer.log`，`cluster_audit/$auditfilename` |
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:83]:
```suggestion
```

不包含 tidb 日志

File: clinic/clinic-collect-sql-query-plan.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+# 要采集的数据类型
+collectors = [
+    "log",
+    "config",
+    "db_vars",
+    "sql_bind",
+    "statistics",
+    "explain"
+]
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:105]:
```suggestion
collectors = [
    "config",
    "sql_bind",
    "plan_replayer"
]
```

File: clinic/clinic-collect-sql-query-plan.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+全量数据采集的输出结果包含以下集群现场信息：
+
+| 序号 | 采集内容 | 调用的 Diag collector | 输出文件 |
+| :--- | :--- | :--- | :--- |
+| 1 | TiDB 配置信息 | `config` | `tidb.toml` |
+| 2 | TiDB Session 系统变量 | `db_vars` | `global_variables.csv`，`mysql.tidb.csv` |
+| 3 | TiDB 执行计划绑定信息 (SQL Binding) | `sql_bind` | `sql_bind/global_bind.csv` |
+| 4 | `sql-statement` 中所包含的表结构 | `statistics` | `statistics/<db.table>.schema` |
+| 5 | `sql-statement` 中所包含表的统计信息 | `statistics` | `statistics/<db.table>.json` |
```
Comment (Yisaer) [inline clinic/clinic-collect-sql-query-plan.md:81]:
```suggestion
| 2 | TiDB Session 系统变量 | `plan_replayer` | `plan_replayer.zip/variables.toml` |
| 3 | TiDB 执行计划绑定信息 (SQL Binding) | `sql_bind` | `sql_bind/global_bind.csv` |
| 4 | `sql-statement` 中所包含的表结构 | `plan_replayer` | `plan_replayer.zip/schema/<db.table>.schema.txt` |
| 5 | `sql-statement` 中所包含表的统计信息 | `plan_replayer` | `plan_replayer.zip/stats/<db.table>.json` |
```

**Review by Yisaer — APPROVED** [review]

**Review by qqqdan — APPROVED** [review]

**Pull request merged**